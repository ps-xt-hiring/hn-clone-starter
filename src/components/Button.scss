@import '../common/variables';
.link-button {
  color:$black;
  background: none;
  border: none;
  margin-left: 5px;
}
