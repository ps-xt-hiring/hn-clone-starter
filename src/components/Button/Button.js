/* eslint-disable */
import React from 'react';
import styles from './Button.module.css';
/**
 * Basic Button Component
 *
 * @param {*} {
 *  clickHandler : ()=>{},
 *  text : String,
 *  type = "button"
 * }
 */
const Button = ({ clickHandler, text, type = 'button' }) => (
  <button className={styles.btn} type={type} onClick={clickHandler}>
    {text}
  </button>
);

export default Button;
