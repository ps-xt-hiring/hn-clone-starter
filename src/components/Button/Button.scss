@import '../../common/colors.scss';

.btn-component {
  background-color: transparent;
  border: none;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  cursor: pointer;
  outline: 0;
}

.secondary {
  color: $black;
}

.primary {
  color: $orange;
}

.other {
  color: $white;
}
