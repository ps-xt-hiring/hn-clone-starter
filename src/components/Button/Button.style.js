import { css } from 'styled-components';
import styles from '../../styles';

export default css`
  &.button {
    background-color: ${styles.colors.white};
    border: 2px solid ${styles.colors.backgroundColorPrimary};
    border-radius: 2px;
    cursor: pointer;
  }
`;
