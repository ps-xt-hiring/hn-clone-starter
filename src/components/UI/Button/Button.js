import React from 'react';

import PropTypes from 'prop-types';
import Classes from './Button.module.scss';

const button = props => (
  <button
    className={[Classes.Button].join(' ')}
    onClick={props.clicked}
    disabled={props.disabled}
    hidden={props.show}
  >
    {props.children}
  </button>
);

button.propTypes = {
  clicked: PropTypes.func,
  disabled: PropTypes.bool,
  hidden: PropTypes.bool,
  children: PropTypes.string
};

export default button;
