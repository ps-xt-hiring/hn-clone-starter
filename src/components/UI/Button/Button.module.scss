@import "../../../sass/abstract/variables";
.Button {
    background-color: transparent;
    border: none;
    color: $header-bg-color;
    cursor: pointer;
    font: inherit;
    padding: 10px;
    margin: .1rem 15%;
    font-size: 1.1rem;
}
