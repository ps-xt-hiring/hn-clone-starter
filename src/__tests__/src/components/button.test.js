import React from 'react'
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Button from '../../../components/module/button'

configure({ adapter: new Adapter() });


describe('Button Component', () => {
    let component;

    beforeEach(() => {
        component = shallow(<Button />)
    })
    it('Component renders', () => {
        expect(component).toBeTruthy();
    })
    it('text show', () => {
        component = shallow(<Button>show</Button>)
        expect(component.props().children).toEqual('show');
    })
    it('text hide', () => {
        component = shallow(<Button>hide</Button>)
        expect(component.props().children).toEqual('hide');
    })
})
