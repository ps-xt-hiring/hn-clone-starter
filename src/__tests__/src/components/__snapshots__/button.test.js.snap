// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders correctly 1`] = `
<button
  className="default"
  onClick={[Function]}
  type="button"
/>
`;
